### PR TITLE
DOC: Improve module docs of matplotlib.scale

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

**Comment by timhoffm** [comment]
Doc build: https://output.circle-artifacts.com/output/job/893aaa5f-4d0c-46f7-bec2-340f50bb79da/artifacts/0/doc/build/html/api/scale_api.html

File: lib/matplotlib/scale.py [hunk 1]
```diff
@@ -1,16 +1,36 @@
 """
 Scales define the distribution of data values on an axis, e.g. a log scaling.
-They are defined as subclasses of `ScaleBase`.
 
-See also `.axes.Axes.set_xscale` and the scales examples in the documentation.
+Scales are subclasses `ScaleBase`. Their mapping is implemented through `.Transform`
```
Comment (jklymak) [inline lib/matplotlib/scale.py:4]: This is pretty technical for right at the top?  Maybe down around the new L26 would be better?

File: lib/matplotlib/scale.py [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+A user will often only use the scale name, e.g. when setting the scale through
+`~.Axes.set_xscale`: ``ax.set_xscale("log")``.
+
+See also the :ref:`scales examples <sphx_glr_gallery_scales>` in the documentation.
+
+Custom scaling can be achieved through `FuncScale`, or by creating your own
+`ScaleBase` subclass (see :doc:`/gallery/scales/custom_scale`).
+Third parties can register their scales by name through `register_scale`.
 
 Matplotlib also supports non-separable transformations that operate on both
```
Comment (jklymak) [inline lib/matplotlib/scale.py:30]: I find this sentence pretty confusing on its own.  Appreciate you are just changing the formatting here, but if we are improving things, we might mention that the previous scales act on one axis at a time, and are for Cartesian axes.

**Review by jklymak — APPROVED** [review]
Couple of small suggestions you can take or leave.

File: lib/matplotlib/scale.py (hunk 2 above)
Comment (timhoffm) [reply lib/matplotlib/scale.py:29]: > , and are for Cartesian axes.

This is not quite true, you can e.g set a log radial axis in a polar plot.

I'm tempted to leave out the paragraph. I think that scales and projections are mostly independent of each other (but am not completely sure).

Comment (jklymak) [reply lib/matplotlib/scale.py:29]: Ok fair enough.  Maybe "scales are typically applied individually to the axis of a separable axes. Non-separable axes are dealt with using projections".  But maybe you are correct that you could do without it here.

Comment (timhoffm) [reply lib/matplotlib/scale.py]: I've removed that paragraph. Irrespective of the exact technical relationship. People looking at scales will typically have a certain projection (mostly cartesian axes) in mind and only consider the scaling behavior along the given axes. It's likely not helpful for them to bring up the topic of other projections here.

**Pull request merged**